### ISSUE TITLE
[feature] Add more malicious SLSA subject layout tests to generate-attestations

### ISSUE THREAD

**Issue by asraa** [slsa-framework/slsa-github-generator#1423]
**Is your feature request related to a problem? Please describe.**
Add strict validation to the SLSA subject layout for generate-attestations. 

Validate that malformed subject layouts cause errors.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by laurentsimon** [comment]
@asraa can you elaborate on "strict validation"?

**Comment by asraa** [comment]
Yes! I'm fairly certain it doesn't even have to be a valid JSON of the type we have - i don't think javascript enforced that. Meaning, we could produce some arbitrary JSON and have that incorporated as the subjects part which would then make it not a valid intoto statement

**Comment by laurentsimon** [comment]
so iiuc, this line https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/generate-attestations/src/attestation.ts#L70-L72 is what needs further validation, correct?

**Comment by laurentsimon** [comment]
Given that subject is defined in https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/generate-attestations/src/intoto.ts#L17-L20, I think validation means:
- verify the hash algorithm 
- verify the length of its value

The name is arbitrary.

Is the above correct?

**Comment by asraa** [comment]
I think it is this entire piece 
https://github.com/slsa-framework/slsa-github-generator/blob/7f9306a166326f33f0aeb3c1a217cedf3fb981c5/.github/actions/generate-attestations/src/attestation.ts#L42

I'm not sure how the type casting to types.Layout works with errors, but basically the idea is that if I add in test cases with completely bogus JSON, will it fail gracefully?

**Comment by asraa** [comment]
Yes - it was more for robustness for the TRW author to see if they are getting an error.
